src/compiler/cpp_generator.cc

```diff
@@ -84,8 +84,8 @@ void PrintIncludes(Printer *printer, const std::vector<grpc::string> &headers,
     }
   }
 
-  for (auto i = headers.begin(); i != headers.end(); i++) {
-    vars["h"] = *i;
+  for (auto i : headers) {
```

Comment:
Suggest using `const auto&` here, to avoid an unnecessary copy.  Same thing in all range-based for loops.

Also suggest renaming this variable from `i` to `header`, since it's no longer an iterator.